File: src/Atlasd/Battlenet/Exceptions/BNFTPProtocolException.cs
using System;

namespace Atlasd.Battlenet.Exceptions
{
    class BNFTPProtocolException : ProtocolException
    {
        public BNFTPProtocolException(ClientState client) : base(Battlenet.ProtocolType.Types.BNFTP, client) { }
        public BNFTPProtocolException(ClientState client, string message) : base(Battlenet.ProtocolType.Types.BNFTP, client, message) { }
        public BNFTPProtocolException(ClientState client, string message, Exception innerException) : base(Battlenet.ProtocolType.Types.BNFTP, client, message, innerException) { }
    }
}

File: src/AtlasV/Battlenet/Exceptions/BNFTPProtocolException.vb
Imports System

Namespace AtlasV.Battlenet.Exceptions
    Class BNFTPProtocolException
        Inherits ProtocolException

        Public Sub New(ByVal varClient As ClientState)
            MyBase.New(Battlenet.ProtocolType.Types.BNFTP, varClient)
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String)
            MyBase.New(Battlenet.ProtocolType.Types.BNFTP, varClient, varMessage)
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String, ByVal varInnerException As Exception)
            MyBase.New(Battlenet.ProtocolType.Types.BNFTP, varClient, varMessage, varInnerException)
        End Sub
    End Class
End Namespace
